commons/src/main/java/org/dhis2/commons/bindings/TEICardExtensions.kt

```diff
@@ -54,42 +58,61 @@ fun List<Program>.addEnrollmentIcons(
     parent: ComposeView,
     currentProgram: String?
 ) {
+    val listSize = this.size
+    var programCount = 0
     parent.apply {
         setContent {
```

Comment:
I think that in this extension function there are mixing concerns as the presentation logic of calculating the icons to draw and number of remaining programs with the UI code. 
In my opinion, to make the function testable (we can create a `TEICardExtensionsTest` to test the function), you can split the extension and the UI returning in the extension function a data class, for example:
`data class EnrollmentIconData(val color: Color, val imageResource: Int)`
And the return the items in the extension function like:
`fun List<Program>.getEnrollmentIconsData(context: Context, currentProgram: String?): List<EnrollmentIconData>`
Now you can write unit test for the logic.
And where you are using the extension function you will receive a list of items to render so you can write the Ui code to render.
What do you think?, let me know if you have any doubt.

Reply:
I agree it is not a good idea to mix the presentation logic with the UI code. I will try and do it as you say. Thanks for the feedback!